### DISCUSSION TITLE
Install FVM in linux machine but downloaded variant is Mac OS

### DISCUSSION THREAD

**Discussion by ranjithbobby89** [Homebrew/discussions#5127] [Linux]
### Additional Info

- [ ] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
>Homebrew bundle config 
HOMEBREW_VERSION: 4.2.8
ORIGIN: https://github.com/Homebrew/brew
HEAD: eb7338abba43112e16e178d4329d5483d6d8fd6f
Last commit: 19 hours ago
Core tap HEAD: ce96caf9bb48f92c40247fe8560c1b2d842acb5f
Core tap last commit: 51 minutes ago
Core tap JSON: 13 Feb 11:00 UTC
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 16
HOMEBREW_NO_INSTALL_CLEANUP: set
HOMEBREW_NO_INSTALL_FROM_API: set
Homebrew Ruby: 3.1.4 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 16-core 64-bit skylake
Clang: N/A
Git: 2.25.1 => /bin/git
Curl: 7.68.0 => /bin/curl
Kernel: Linux 5.10.192-183.736.amzn2.x86_64 x86_64 GNU/Linux
OS: Ubuntu 20.04.3 LTS (focal)
Host glibc: 2.31
/usr/bin/gcc: 9.3.0
/usr/bin/ruby: 2.7.0
glibc: 2.35_1
gcc@11: N/A
gcc: 13.2.0
xorg: N/A
```

### Output of `brew doctor`

```shell
>Homebrew bundle Doctor
Your system is ready to brew.
```

### Description of issue

Tried to install fvm using homebrew but I noticed installed sdk variant was Mac OS so failed to install

>Homebrew bundle install
/home/linuxbrew/.linuxbrew/bin/brew tap homebrew/bundle
==> Tapping homebrew/bundle
Cloning into '/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-bundle'...
Tapped 1 command (107 files, 2.2MB).
Tapping leoafarias/fvm
Installing leoafarias/fvm tap. It is not currently installed.
==> Tapping leoafarias/fvm
Cloning into '/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/leoafarias/homebrew-fvm'...
Tapped 9 formulae (22 files, 56KB).
Installing fvm
Installing fvm formula. It is not currently installed.
==> Fetching leoafarias/fvm/fvm
**==> Downloading https://storage.googleapis.com/dart-archive/channels/stable/release/3.2.6/sdk/dartsdk-macos-x64-release.zip**
==> Downloading https://github.com/leoafarias/fvm/archive/3.0.1.tar.gz
==> Downloading from https://codeload.github.com/leoafarias/fvm/tar.gz/refs/tags/3.0.1
==> Installing fvm from leoafarias/fvm
==> /home/linuxbrew/.linuxbrew/Cellar/fvm/3.0.1/libexec/bin/dart pub get
Last 15 lines from /root/.cache/Homebrew/Logs/fvm/01.dart:
2024-02-13 10:29:10 +0000
/home/linuxbrew/.linuxbrew/Cellar/fvm/3.0.1/libexec/bin/dart
pub
get
/home/linuxbrew/.linuxbrew/Cellar/fvm/3.0.1/libexec/bin/dart: 1: ��������H__PAGEZEROh__TEXT�#�#: not found
/home/linuxbrew/.linuxbrew/Cellar/fvm/3.0.1/libexec/bin/dart: 2: Syntax error: Unterminated quoted string
If reporting this issue please do so at (not Homebrew/brew or Homebrew/homebrew-core):
  https://github.com/leoafarias/homebrew-fvm/issues
These open issues may also help:
leoafarias/fvm/fvm 2.4.1 did not build https://github.com/leoafarias/homebrew-fvm/issues/15
Can't install fvm on linuxbrew https://github.com/leoafarias/homebrew-fvm/issues/17
Can't install fvm by brew https://github.com/leoafarias/homebrew-fvm/issues/16
fvm relying on an older version of dart and failing via brew install https://github.com/leoafarias/homebrew-fvm/issues/14
Installing fvm has failed!
Homebrew Bundle failed! 1 Brewfile dependency failed to install.

why this discrepancy please help to fix and update 

Thanks in Advance!

**Comment by ZhongRuoyu** [comment]
Your `dart-sdk` resource is defined [this way](https://github.com/leoafarias/homebrew-fvm/blob/b154ded00d2221f90e6176cdfb0dfa647b20e0bd/fvm.rb#L10-L22), which only takes care of macOS:

```ruby
  # Determine architecture and set the Dart SDK resource accordingly
  dart_sdk_url, dart_sdk_sha = if Hardware::CPU.intel?
    ["https://storage.googleapis.com/dart-archive/channels/stable/release/3.2.6/sdk/dartsdk-macos-x64-release.zip",
     "97661f20230686381f4fc5b05a63c6b5d5abc9570bf93ad4e5fc09309cd98517"]
  elsif Hardware::CPU.arm?
    ["https://storage.googleapis.com/dart-archive/channels/stable/release/3.2.6/sdk/dartsdk-macos-arm64-release.zip",
     "2e04c91039f9cc05b2e93ce788aa1ce08bc4df5b50c84d6b4e21ba2b2538adb6"]
  end

  resource "dart-sdk" do
    url dart_sdk_url
    sha256 dart_sdk_sha
  end
```

You need to adjust the formula to make sure the resource works on Linux too. See [`rust.rb`](https://github.com/Homebrew/homebrew-core/blob/ce96caf9bb48f92c40247fe8560c1b2d842acb5f/Formula/r/rust.rb#L47-L69) for example.

**Comment by p-linnane** [comment]
This is also from a third-party tap, not our main Homebrew provided ones. From the output you posted:
>If reporting this issue please do so at (not Homebrew/brew or Homebrew/homebrew-core):
https://github.com/leoafarias/homebrew-fvm/issues
